reference/forms/types/choice.rst

```diff
@@ -147,7 +147,7 @@ Field Options
 choices
 ~~~~~~~
 
-**type**: ``array`` **default**: ``[]``
```

Comment:
Callable is not allowed in `ChoiceType` option resolver

https://github.com/symfony/symfony/blob/db8b9da89978110c7b1b299e2218dbe6abcbef4e/src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php#L380

```php
$resolver->setAllowedTypes('choices', ['null', 'array', \Traversable::class]);
```

What is the return type of `MyEnum::myChoices()` in your comment ?

---

https://github.com/symfony/symfony/blob/db8b9da89978110c7b1b299e2218dbe6abcbef4e/src/Symfony/Component/Form/Extension/Core/Type/EnumType.php#L32C95-L32C100

```php
 ->setDefault('choices', static fn (Options $options): array => $options['class']::cases())
```

Reply:
If I understand what you did here correctly, your configuration is relying on the built-in feature of the OptionsResolver component to define default options depending on another option: https://symfony.com/doc/current/components/options_resolver.html#default-values-that-depend-on-another-option

I don't think we should cover this by adding `callable` everywhere as the allowed type.

Reply:
Re-reading your PR description I am not sure you really do pass a callback to `choices` option.

```php
->add('foo', EnumType::class, [
    'choices' => MyEnum::myChoices(),
])
```

Is this how you do it?

Reply:
By the way, I submitted a PR that is supposed to fix the issue you reported. Please give it a try. :)